app/logic/Calendar/Event.ts

```diff
@@ -729,6 +729,7 @@ export class Event extends Observable {
    * Used when a master recurrence rule is removed or changed incompatibly.
    */
   protected clearExceptions() {
+    assert(this.recurrenceCase == RecurrenceCase.Master, "Must be a recurring master");
```

Comment:
`this.recurrenceCase` can't be relied upon at this point, because we might have just cleared the recurrence rule.